### ISSUE TITLE
[HOLD for payment 2023-05-03] [$1000] The cursor is not displayed as a pointer/hand on the pressable date of birth field

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#16513]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to www.staging.new.expensify.com 
3. Go to Settings > Profile > Personal details 
4. Go to Date of birth of an account where the DOB has not yet been entered 
5. Hover over the Date field

## Expected Result:
Pointer/hand cursor should be displayed since it is a pressable field

## Actual Result:
Text insertor cursor is displayed

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.89-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/228079056-fb21fbc1-f73f-4a68-a268-96af449f17df.mov

https://user-images.githubusercontent.com/43996225/227731298-1825a2b2-0b25-425a-90e1-3a6200c44f28.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @adeel0202 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1679746931142549

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017c7c93baa6275770</li>
        <li>Upwork Job ID: 1646542054026485760</li>
        <li>Last Price Increase: 2023-04-13</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @laurenreidexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by laurenreidexpensify** [comment]
I cannot reproduce this on staging or prod, closing. I recorded this but don't want to expose my PII in a video here, so not going to upload the evidence.

**Comment by kavimuru** [comment]
@laurenreidexpensify  My bad. I have uploaded correct video. Could you please check again?

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~017c7c93baa6275770

**Comment by MelvinBot** [comment]
Current assignee @laurenreidexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by dhairyasenjaliya** [comment]
### Proposal

**Please re-state the problem that we are trying to solve in this issue.**
 
 - The cursor is not displayed as a pointer/hand on the pressable date of birth field
 
**What is the root cause of that problem?**

- The reason the current cursor is the default Text insertor cursor because we are using `<Textinput>` to display the selected date which by default has a cursor of Text insertor
 
**What changes do you think we should make in order to solve the problem?**

- We need to override the default cursor to show as a `pointer` on hover 

- We just need to add `styles.cursorPointer` as `inputStyle` props  into `<Textinput>` which is in below component

https://github.com/Expensify/App/blob/3a9ce6fc59bb54d25749579be9b082c7600b235b/src/components/NewDatePicker/index.js#L129

- It will resolve the issue at global level for `NewDatePicker` component 
 
 **Changes**
 
 ```diff
 
  <TextInput                    
+     inputStyle={[styles.cursorPointer]}
 />
 
 ```
 
 
**What alternative solutions did you explore? (Optional)**

- We could also change `<Textinput>` to `<TouchableWithoutFeedback>` or `<View>` to display selected date

**Comment by MelvinBot** [comment]
Triggered auto assignment to @MariaHCD (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by Santhosh-Sellavel** [comment]
The calendar Icon or Field is not clickable, just a read-only, 

Do we need the Icon here? 

https://user-images.githubusercontent.com/85645967/232132609-ea170f6a-13ab-4d95-82be-12db31bbf332.mov

@MariaHCD thoughts?

**Comment by MelvinBot** [comment]
@MariaHCD @laurenreidexpensify @Santhosh-Sellavel this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a> and will need to go internal. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by MariaHCD** [comment]
Thanks for the proposals here! Looks like the OP is a bit outdated since the calendar is now always open by default.

So it seems we have two options here:
1. Don't have the cursor change into the hand pointer when hovering over the date icon OR
2. Get rid of the date icon completely since we can't really toggle the date picker (so perhaps it doesn't really serve a purpose now?)

Does that sound correct, @Santhosh-Sellavel?

cc: @Beamanator (since you've worked on this date picker) @shawnborton (for your thoughts on the UI/UX here!)

**Comment by mountiny** [comment]
Yeah that sounds good to me, @MariaHCD do you want me to take over while you ooo?

**Comment by mountiny** [comment]
@Santhosh-Sellavel Which proposal should we go ahead with if we agree on:

> Don't have the cursor change into the hand pointer when hovering over the date icon

**Comment by mountiny** [comment]
@Santhosh-Sellavel whats the recommendation here given we know the expected behaviour now?

**Comment by Santhosh-Sellavel** [comment]
We have suggestions from @Prince-Mendiratta about refactoring and but the solution I like @daraksha-dk's idea from [here](https://github.com/Expensify/App/issues/16513#issuecomment-1507208419)

i.e
> Another way to deal with this is using disabled prop to get the default cursor on the Input and add cursor style conditionally for Input Icon based on this.props.disabled

your thoughts @mountiny ?

**Comment by mountiny** [comment]
@Santhosh-Sellavel thanks, I guess then we can go wit @daraksha-dk proposal then

**Comment by mountiny** [comment]
I guess it would be also nice to have, but only one person is assigned to a job so either we can create some refactoring job/issue if its worth it or do nothign for now. @Santhosh-Sellavel do you think its valuable refactoring for now?

**Comment by laurenreidexpensify** [comment]
@Prince-Mendiratta @daraksha-dk are we good to move forward here? Is there a PR up yet?

**Comment by Santhosh-Sellavel** [comment]
@shawnborton we have inconsistent font color here for the disabled state should we do something about it?

 I'm okay with the other proposal as well, cc: @mountiny thoughts?

**Comment by shawnborton** [comment]
I don't think the input text should appear disabled, we should just use our standard text color there.

**Comment by mountiny** [comment]
agree, it should not appear disabled

**Comment by Santhosh-Sellavel** [comment]
Okay we just change the cursor style here

**Comment by mountiny** [comment]
Merged

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.5-6 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17783


If no regressions arise, payment will be issued on 2023-05-03. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@Santhosh-Sellavel] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@Santhosh-Sellavel] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@Santhosh-Sellavel] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@Santhosh-Sellavel] Determine if we should create a regression test for this bug.
- [ ] [@Santhosh-Sellavel] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@laurenreidexpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by laurenreidexpensify** [comment]
Looks like this is eligible for 3 day bonus. 

Offers sent to @adeel0202 @Santhosh-Sellavel @Prince-Mendiratta @daraksha-dk 

I have split up the $1500 fee for fixing ($1000) and 3 day merge bonus ($500) for @Prince-Mendiratta and @daraksha-dk to $750 each

**Comment by laurenreidexpensify** [comment]
Paid @adeel0202 @Prince-Mendiratta so far

**Comment by Santhosh-Sellavel** [comment]
I think this is an improvement and won't need regression testing thoughts. @mobilify
